dot/sync/chain_sync.go

```diff
@@ -932,6 +936,44 @@ func (cs *chainSync) validateJustification(bd *types.BlockData) error {
 	return nil
 }
 
+func (cs *chainSync) getHighestBlock() (int64, error) {
+	cs.RLock()
+	ps := cs.peerState
+	cs.RUnlock()
+
+	if len(ps) <= 0 {
+		return 0, errNoPeers
+	}
+
+	// improve performance returning first highestBlock
+	if len(ps) == 1 {
+		for _, p := range ps {
+			if p.number == nil {
+				return 0, errNilBlockData
+			}
+			return p.number.Int64(), nil
+		}
+	}
+
+	numbers := []*big.Int{}
+	for _, ps := range ps {
+		if ps.number == nil {
+			continue
+		}
+		numbers = append(numbers, ps.number)
+	}
+
+	if len(numbers) <= 0 {
+		return 0, errNilBlockData
+	}
+
+	sort.SliceStable(numbers, func(i, j int) bool {
+		return numbers[i].Cmp(numbers[j]) >= 0 // numbers[i] >= numbers[j]: https://pkg.go.dev/math/big#Int.Cmp
+	})
```

Comment:
Thinking better... Instead of sorting it, we can just compare inside of the map loop and get the max value from there. What you think? In this way, we wont need `numbers` anymore. 🚀

Reply:
@EclesioMeloJunior @qdm12 

As I have commented, I changed the logic of this function. I built this Benchmark to check which logic is better:

```golang
func BenchmarkChainSync_highestBlock(b *testing.B) {
	type input struct {
		peerState map[peer.ID]*peerState
	}
	type output struct {
		highestBlock int64
		err          error
	}
	type test struct {
		name string
		in   input
		out  output
	}
	m := make(map[peer.ID]*peerState)
	for i := 0; i < 5000; i++ {
		id := uuid.New()
		n := rand.Intn(20000)
		ps := &peerState{
			who:    "idtest#1",
			hash:   common.BytesToHash([]byte("hash#1")),
			number: big.NewInt(int64(n)),
		}
		m[peer.ID(id.String())] = ps
	}
	tests := []test{
		{
			name: "when has two peers (p1, p2) with p1.number 190 and p2.number 90 should return 190, nil",
			in: input{
				peerState: m,
			},
			out: output{
				highestBlock: int64(193),
				err:          nil,
			},
		},
	}
	cs := &chainSync{}

	for _, ts := range tests {
		b.Run(ts.name, func(b *testing.B) {
			cs.peerState = ts.in.peerState
			cs.getHighestBlock()
		})
	}
}
```

* Result:
```
# Before

BenchmarkChainSync_highestBlock/when_has_two_peers_(p1,_p2)_with_p1.number_190_and_p2.number_90_should_return_190,_nil-8         	1000000000	         0.002441 ns/op	       0 B/op	       0 allocs/op

# After

BenchmarkChainSync_highestBlock/when_has_two_peers_(p1,_p2)_with_p1.number_190_and_p2.number_90_should_return_190,_nil-8         	1000000000
```

As you can see, we can get a better performance using the "After" approach. For that, I would like a second review here: 8b40b9b

Reply:
Sounds good!